narwhal/worker/src/batch_maker.rs

```diff
@@ -129,11 +130,15 @@ impl BatchMaker {
 
                         timer.as_mut().reset(Instant::now() + self.max_batch_delay);
                         self.batch_start_timestamp = Instant::now();
+
+                        // Yield once per size threshold to allow other tasks to run.
+                        tokio::task::yield_now().await;
```

Comment:
Have we tried removing completely and benchmark? I mean, do we have real concerns that batch maker can really cause some starvation here?

Reply:
Yes, I have experimented with removing this completely and it resulted in less spikes on the backlog of `tx_batch_maker`. There can be noises but I'm reasonably sure not yielding per transaction is improving the health of workers. I still kept the logic to yield when a batch is full, just to avoid breaking the case when worker is receiving a lot transactions.